### DISCUSSION TITLE
Help: Using Perplexity API with the 3rd party provider option Always Get Null Return

### DISCUSSION THREAD

**Discussion by siyangni** [logancyang/obsidian-copilot#674] [Q&A] [answered]
I tried to use Perplexity.ai API with the 3rd-party function, with Base URL = https://api.perplexity.ai, Model Name = llama-3.1-sonar-small-128k-chat, API Key = {my API key}. But I always got null return. I also tried other model names provided on the official Doc of Perplexity and got the same null return all the time. 

On the Doc (https://docs.perplexity.ai/guides/getting-started), Perplexity said "The API is conveniently OpenAI client-compatible for easy integration with existing applications." Would you investigate this issue when you have time? Thank you!

**Comment by logancyang** [comment, marked as answer]
Please check: https://www.obsidiancopilot.com/en/docs/settings#adding-custom-models

**Reply by siyangni** [reply]
Brilliant!
